### PR TITLE
Enable VB emit tests on CoreClr

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-compiler for review

File: src/Compilers/VisualBasic/Test/Emit/Emit/FieldInitializerTests.vb [hunk 1]
```diff
@@ -583,6 +583,7 @@ BC30424: Constants must be of an intrinsic or enumerated type, not a class, stru
 </expected>)
         End Sub
 
+#If NET46 Then
```
Comment (333fred) [inline src/Compilers/VisualBasic/Test/Emit/Emit/FieldInitializerTests.vb]: Is there some type here that causes the test to be unable to be compiled on coreclr? If not, why not use a ConditionalFact?

File: src/Compilers/VisualBasic/Test/Emit/Emit/OptionalArgumentsTests.vb [hunk 2]
```diff
@@ -814,7 +814,8 @@ Class P
         Report(C.F6())
     End Sub
     Shared Sub Report(o As Object)
-        System.Console.WriteLine("{0}: {1}", o.GetType(), o)
+        Dim value As Object = If (TypeOf o is Date, DirectCast(o, Date).ToString("MM/dd/yyyy HH:mm:ss"), o)
```
Comment (333fred) [inline src/Compilers/VisualBasic/Test/Emit/Emit/OptionalArgumentsTests.vb]: This is why I suggested moving the datetime thing to a helper

**Comment by 333fred** [comment]
@jaredpar looks like the coreclr tests aren't happy with this change

**Comment by jaredpar** [comment]
@333fred 

> looks like the coreclr tests aren't happy with this change

Weird. They are passing locally. Wonder if my last rebase broke something.

**Comment by jaredpar** [comment]
Nope they do indeed all pass locally. This should be fun.

**Comment by jaredpar** [comment]
retest this please

File: src/Compilers/VisualBasic/Test/Emit/ExpressionTrees/ExpTreeTestResources.vb [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+Option Strict On
+Option Explicit On
+
+Imports System
+Imports System.IO
+Imports System.Reflection
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests
+
+    Public Class ExpTreeTestResources
```
[the remark this answers is not in the document]
Comment (jaredpar) [reply src/Compilers/VisualBasic/Test/Emit/ExpressionTrees/ExpTreeTestResources.vb]: Think we need deeper integration. Have to verify that we don't accidentally hit this bug again: using a RESX in a multi-targetted project.

**Pull request merged**